libraries/js/text/regex.effekt

```diff
@@ -1,33 +1,221 @@
 module text/regex
 
 import string
+import option
+import list
+
 
 extern type Regex
 
-record Match(matched: String, index: Int)
+// matches and groups
+record Range(start: Int, end: Int)
+record Group(content: String, index: Option[Range])
+record Match(content: String, start: Int, end: Int,  groups: List[Group])
```

Comment:
👀 Renaming the first field of `Match` to `content` is a change that ripples through the other backends.
For example in the Chez backends, the `Match` type is unchanged, which triggers the test failures you're seeing:
https://github.com/effekt-lang/effekt/blob/cd18ad896b9f1c856a9d6002fd0a5733e5cb9a15/libraries/chez/common/text/regex.effekt#L9

(and the same API is probably [?] assumed for the ML backend:)
https://github.com/effekt-lang/effekt/blob/cd18ad896b9f1c856a9d6002fd0a5733e5cb9a15/libraries/ml/text/regex.effekt#L36

WDYT is the best step forward? I can see two straightforward ways here:
- rename the `matched` field in all backends to `content`
- rename the `content` field(s) here to `matched` to align with the other backends

(I'm fine with both of these solutions, I think)

Reply:
_Side note: Let me note again that ultimately we will drop Chez and ML._

Reply:
I'll go with the second option :) Matched is still expressive enough provide a good user experience